core/domain/blog_domain.py

```diff
@@ -0,0 +1,609 @@
+                    'received: \'%s\'' % tag)
+
+            if (tag[0] not in string.ascii_lowercase or
+                    tag[-1] not in string.ascii_lowercase):
+                raise utils.ValidationError(
+                    'Tags should not start or end with whitespace, received: '
+                    '\'%s\'' % tag)
+
+            if re.search(r'\s\s+', tag):
+                raise utils.ValidationError(
+                    'Adjacent whitespace in tags should be collapsed, '
+                    'received: \'%s\'' % tag)
+
+        if strict:
+            if len(tags) == 0:
+                raise utils.ValidationError(
+                    'Atleast one tag should be selected')
+
+        if len(set(tags)) != len(tags):
+            raise utils.ValidationError(
+                'Some tags duplicate each other')
+
+    @classmethod
+    def require_valid_title(cls, title, strict):
+        """Checks whether the blog post title is a valid one.
+
+        Args:
+            title: str. The title to validate.
+            strict: bool. Enable strict checks on the blog post when the blog
+                post is published or is going to be published.
+        """
+
+        if not isinstance(title, python_utils.BASESTRING):
+            raise utils.ValidationError('Title should be a string.')
+
+        title_limit = constants.MAX_CHARS_IN_BLOG_POST_TITLE
+        if len(title) > title_limit:
+            raise utils.ValidationError(
+                'Blog Post title should at most have %d chars, received: %s'
+                % (title_limit, title))
+
+        if strict:
+            if len(title) == 0:
+                raise utils.ValidationError('Title should not be empty')
+            if not re.match(constants.VALID_BLOG_POST_TITLE_REGEX, title):
+                raise utils.ValidationError(
+                    'Title field contains invalid characters. Only words'
+                    '(a-zA-Z) separated by spaces are allowed. Received %s'
+                    % title)
+
+    @classmethod
+    def require_valid_url_fragment(cls, url_fragment):
+        """Checks whether the url fragment of the blog post is a valid one.
+
+        Args:
+            url_fragment: str. The url fragment to validate.
+        """
+
+        url_limit = constants.MAX_CHARS_IN_BLOG_POST_URL_FRAGMENT
+        utils.require_valid_url_fragment(
+            url_fragment, 'Blog Post URL Fragment', url_limit)
+
+    def to_dict(self):
+        """Returns a dict representing this blog post domain object.
+
+        Returns:
+            dict. A dict, mapping all fields of blog post instance.
+        """
+        return {
+            'id': self.id,
+            'author_name': user_services.get_user_id_from_username(
+                self.author_id),
+            'title': self.title,
+            'content': self.content,
+            'thumbnail_filename': self.thumbnail_filename,
+            'tags': self.tags,
+            'url_fragment': self.url_fragment,
```

Comment:
tod_dict method is mostly used to send data to the frontend, in that case, should we also add `published_on` and `last_updated` information? [not sure!] (I see this information in Blog posts.)

Reply:
Thanks. Yup it will be required. Added.